pytorch_lightning/callbacks/base.py

```diff
@@ -57,7 +57,12 @@ def _generate_state_key(self, **kwargs: Any) -> str:
         return f"{self.__class__.__qualname__}{repr(kwargs)}"
 
     def on_configure_sharded_model(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
-        """Called before configure sharded model."""
+        r"""
+        ..deprecated:: v1.6
```

Comment:
The deprecation warning should come after the main docstring

Reply:
I don't think that's how it is done at other places. For eg., on_init_start/on_init_end displays the deprecation warning first.